rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -59,19 +59,23 @@ namespace rsx
 			rsx->ctrl->ref.exchange(arg);
 		}
 
-		void semaphore_acquire(thread* rsx, u32 _reg, u32 arg)
+		void semaphore_acquire(thread* rsx, u32 /*_reg*/, u32 arg)
 		{
 			rsx->sync_point_request = true;
 			const u32 addr = get_address(method_registers.semaphore_offset_406e(), method_registers.semaphore_context_dma_406e());
-			if (vm::read32(addr) == arg) return;
+
+#ifdef IS_LE_MACHINE
```

Comment:
Forgot to complain... That's not a sort of loop to benefit from removing one 1-cycle byteswap op, especially with so much code madness.